lib/auth/auth_with_roles.go

```diff
@@ -2648,3 +2652,16 @@ func NewAdminAuthServer(authServer *Server, sessions session.Service, alog event
 		sessions:   sessions,
 	}, nil
 }
+
+// verbsForResourceWithOrigin determines the verbs/actions required of a role
+// to replace the resource currently stored in the backend with the candidate.
+func verbsForResourceWithOrigin(stored, candidate types.ResourceWithOrigin) []string {
+	verbs := []string{types.VerbUpdate}
+	if stored.Origin() == types.OriginConfigFile {
+		verbs = append(verbs, types.VerbCreate)
+	}
+	if candidate.Origin() == types.OriginDefaults {
+		verbs = append(verbs, types.VerbDelete)
+	}
+	return verbs
+}
```

Comment:
The `VerbDelete` condition is probably going away.

However we really need a distinguished privilege (whether `VerbCreate` or not) to prevent mere `VerbUpdate` users from using `tctl create --force --confirm` and thus overwriting (even if just temporarily) the config-file configuration. There should be a straightforward way to make the config file authoritative once it's specified.

I think it's okay for the verbs to have a slightly shifted meaning in the context of singleton resources (that don't really get created or deleted).

Reply:
modifying `AuthPreference` in any way seems like a privileged operation, whether it's replacing a static or dynamic value.

Reply:
Yes, that's why `VerbUpdate` would be always needed.

However, replacing a piece of static configuration should require even more privileges: e.g. `VerbCreate` in addition to the "basic" `VerbUpdate` as proposed here.

BTW, the convention with our handlers is to require both `VerbCreate` and `VerbUpdate` for a set operation.

Reply:
Alright, please remove `VerbDelete` though